libraries/c_sdk/standard/mqtt/README.md

```diff
@@ -1,18 +1,71 @@
 
 
-**Note:** The config `IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES` to override serializer APIs is not supported in this MQTT shim implementation. [MQTT over BLE](https://docs.aws.amazon.com/freertos/latest/userguide/ble-demo.html#ble-demo-mqtt) is redesigned to avoid the dependency on this config by using a [BLE transport implementation](../ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c).
+**Note:** The config `IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES` to override serializer APIs is not supported in this MQTT library for backward compatibility for MQTT V2.x.x APIs. [MQTT over BLE](https://docs.aws.amazon.com/freertos/latest/userguide/ble-demo.html#ble-demo-mqtt) is redesigned to avoid the dependency on this config by using a [BLE transport implementation](../ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c).
+
+## Code size of MQTT library for backward compatibility for MQTT V2.x.x APIs
+
+Code size for MQTT library for backward compatibility for MQTT V2.x.x APIs is calculated using GCC for ARM cortex-M4. Build for calculating code size is done using build configuration for [cypress CY8CKIT-064S0S2-4343W](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_cypress_psoc64.html) by using the `Release` configuration. All the logs were disabled in this build. Refer to the table below for calculated code sizes.
+
+| File | With -O1 Optimization | With -Os Optimization |
+| :-: | :-: | :-: |
+| iot_mqtt_agent.c | 1.3K | 1.1K |
+| iot_mqtt_api.c | 3.8K | 3.3K |
+| iot_mqtt_context_connection.c | 0.2K | 0.2K |
+| iot_mqtt_managed_function_wrapper.c | 0.9K | 0.9K |
+| iot_mqtt_mutex_wrapper.c | 0.4K | 0.3K |
+| iot_mqtt_network.c | 1.2K | 1.1K |
+| iot_mqtt_operation.c | 2.6K | 2.4K |
+| iot_mqtt_publish_duplicates.c | 0.2K | 0.2K |
+| iot_mqtt_serializer_deserializer_wrapper.c | 2.0K | 1.8K |
+| iot_mqtt_subscription.c | 1.3K | 1.2K |
+| iot_mqtt_subscription_container.c | 0.6K | 0.6K |
+| iot_mqtt_validate.c | 0.5K | 0.4K |
+| **Total estimate** | **15K** | **13.5K** |
+
+MQTT library for backward compatibility for MQTT V2.x.x APIs is implemented using [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md). In addition to that, this library for backward compatibility for MQTT V2.x.x APIs also maintains all the dependencies of MQTT V2.x.x library. In order to account for the total code size cost of this library, the code sizes for the dependencies also need to be considered. Please find the code sizes for dependencies below.
+
+1. **coreMQTT library** : MQTT library for backward compatibility for MQTT V2.x.x APIs is implemented using [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md). Memory estimates for [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md) can be found in the documentation [here](https://freertos.org/mqtt/index.html).
```

Comment:
This sounds confusing "MQTT library for backward compatibility"
Are we saying "MQTT compatibility layer for MQTT V2.x.x API " ? I think term library confsuing. We should use term 'compatibility layer' not 'library'

Reply:
Agree.  I will update the term. Thanks